src/hooks/utility.rs

```diff
@@ -92,7 +97,44 @@ pub async fn process_utility_hook(
 }
 
 fn is_support_utility(stmt_type: NodeTag) -> bool {
-    stmt_type == pg_sys::NodeTag::T_ExplainStmt
+    stmt_type == pg_sys::NodeTag::T_ExplainStmt || stmt_type == pg_sys::NodeTag::T_ViewStmt
+}
+
+fn view_query(query_string: &core::ffi::CStr, stmt: *mut pg_sys::ViewStmt) -> Result<bool> {
+    // Get the current schema in Postgres
+    let current_schema = get_postgres_current_schema();
+    // Set DuckDB search path according search path in Postgres
+    set_search_path_by_pg()?;
+
+    let query = unsafe { (*stmt).query as *mut pg_sys::SelectStmt };
+    let from_clause = unsafe { (*query).fromClause };
+    unsafe {
+        let elements: *mut pg_sys::ListCell = (*from_clause).elements;
+        for i in 0..(*from_clause).length {
+            let rv = (*elements.offset(i as isize)).ptr_value as *mut pg_sys::RangeVar;
+
+            // if the schema name is not provided, the current schema is used
+            let relation_name = if (*rv).relname.is_null() {
+                current_schema.as_str()
+            } else {
+                CStr::from_ptr((*rv).relname).to_str()?
+            };
+            let schema_name = if (*rv).schemaname.is_null() {
+                current_schema.as_str()
+            } else {
+                CStr::from_ptr((*rv).schemaname).to_str()?
+            };
+
+            if !view_exists(relation_name, schema_name)? {
```

Comment:
I think we should perform the pushdown checks following the executor hook(check postgres catalog), and the existence of foreign tables in DuckDB should not be a concern here.

1. pushdown checks in utility hook(DDLs, like `prepare`, `explain`, `create view`)  and executor hook acutually are the same.( I think this approach can reduce the occurrence of bugs.)
2. Foreign table records in PG should also be in DuckDB (they may be lazy-loaded). I think we can implement this later.

Reply:
I don't fully understand your approach. Why can't we check `create view` in the utility hook?

Reply:
I think we should follows
the same rules to check those pushdown query(by their plans or...). like  create view as select, the select part should do the same check as in executor hook.
if users restart a pg instance,and create a view. it should be pushdown successfully because foreign tables have been created before.

Reply:
😂 I just think this approach provides better consistency and reduces bugs and confusing issues.（we should follows pg's catalog first i think）